### PR TITLE
FEATURE: SD-857 Handle filter for Geo chart on KD

### PR DESCRIPTION
**Pull request by ptquang86**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [ ] Squash commits 

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2857
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2599

# Related Jira tasks
- SD-857: https://jira.intgdc.com/browse/SD-857

### CODE REVIEW

**Comment by ptquang86** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1420.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6384/ : SUCCESS in 11m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2145/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2146/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2709/ : SUCCESS in 23m 19s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3301/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2147/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2710/ : UNSTABLE in 20m 51s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3302/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2151/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2711/ : SUCCESS in 15m 19s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3318/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2185/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2728/ : SUCCESS in 17m 52s

File: src/execution/fixtures/ExecuteAfm.fixtures.ts [hunk 1]
```diff
@@ -932,13 +932,300 @@ const twoMeasuresOneDimensionResponse: Execution.IExecutionResponses = {
...
+                                identifier: "attr.ziplatlong.state",
+                            },
+                        },
+                    },
+                ],
+            },
+        ],
+        links: {
+            executionResult:
+                "/gdc/app/projects/projectId/executionResults/6726824952228314112?c=5b9c03529139801e7cc5f21c27cd3218&q=eAGlksFPwjAUxv%2BVpbsubJAojMR4UCRcNFE5kR66tcOZdq%2B2r7pJ%2BN9tgUWCFwnX1%2B9973u%2Fvg0x%0AQoPBR6YEmZJlgzVKwUlCSpBONZZMVxvCa6sl6x7AqAX3snTNy1TxtPuwuq3WWQs8H46At9ffIDtg%0AV66cZHpUplC8p5OhdyuYFTMplGhw%2Bbw412Kcp2LfbG9rfuP9GKKpC4fi%2FDzj3PdbpxQznQ9ybPYa%0Alve1%2Bewp0s6%2B6brxzzskf6ud3knv7wZrAYODFJDJwGxFkxWl22Tzm7Q3t8hQnEa4iPB45O0uJDw8%0AJdxv3ce9hHi4AOzR0C1NiBL%2B%2F8r9dVX%2BrBh6wHESx9kgy46YvzgVQRVxiyTA%2FIf0k0knSJhh4CsM%0AOMyaG3Ca0O0Pg1v8Kw%3D%3D%0A&offset=0%2C0&limit=1000%2C1000&dimensions=2&totals=0%2C0",
```
Comment (naibinh) [inline src/execution/fixtures/ExecuteAfm.fixtures.ts]: Can we reduce this long url to be same as [other results](https://github.com/gooddata/gooddata-react-components/blob/188d07ec77a2bcfa0a3f0e9f4ae6167bafd23259/src/execution/fixtures/ExecuteAfm.fixtures.ts#L35)?

File: src/execution/fixtures/ExecuteAfm.fixtures.ts [hunk 2]
```diff
@@ -932,13 +932,300 @@ const twoMeasuresOneDimensionResponse: Execution.IExecutionResponses = {
...
+                                identifier: "attr.ziplatlong.state",
+                            },
+                        },
+                    },
+                ],
+            },
+        ],
+        links: {
+            executionResult:
+                "/gdc/app/projects/projectId/executionResults/8917638730922032128?c=5b9c03529139801e7cc5f21c27cd3218&q=eAGlksFPwjAUxv%2BVpbsubJAojMR4UCRcNFE5kR66tcOZdq%2B2r7pJ%2BN9tgUWCFwnX1%2B9973u%2Fvg0x%0AQoPBR6YEmZJlgzVKwUlCSpBONZZMVxvCa6sl6x7AqAX3snTNy1TxtPuwuq3WWQs8H46At9ffIDtg%0AV66cZHpUplC8p5OhdyuYFTMplGhw%2Bbw412Kcp2LfbG9rfuP9GKKpC4fi%2FDzj3PdbpxQznQ9ybPYa%0Alve1%2Bewp0s6%2B6brxzzskf6ud3knv7wZrAYODFJDJwGxFkxWl22Tzm7Q3t8hQnEa4iPB45O0uJDw8%0AJdxv3ce9hHi4AOzR0C1NiBL%2B%2F8r9dVX%2BrBh6wHESx9kgy46YvzgVQRVxiyTA%2FIf0k0knSJhh4CsM%0AOMyaG3Ca0O0Pg1v8Kw%3D%3D%0A&offset=0%2C0&limit=1000%2C1000&dimensions=2&totals=0%2C0",
```
Comment (naibinh) [inline src/execution/fixtures/ExecuteAfm.fixtures.ts]: Can we reduce this long url to be same as [other results](https://github.com/gooddata/gooddata-react-components/blob/188d07ec77a2bcfa0a3f0e9f4ae6167bafd23259/src/execution/fixtures/ExecuteAfm.fixtures.ts#L35)?

File: src/components/core/base/GeoValidatorHOC.tsx [hunk 3]
```diff
@@ -38,10 +38,10 @@ export function geoValidatorHOC<T>(InnerComponent: React.ComponentClass<T>): Rea
         }
 
         public shouldComponentUpdate(nextProps: IGeoValidatorProps): boolean {
-            const { config } = this.props;
-            const { config: nextConfig } = nextProps;
+            const { config, dataSource } = this.props;
+            const { config: nextConfig, dataSource: nextDataSource } = nextProps;
             // check if buckets, filters and config are changed
-            return !isEqual(config, nextConfig);
+            return !isEqual(config, nextConfig) || !isEqual(dataSource.getAfm(), nextDataSource.getAfm());
```
[the remark this answers is not in the document]
Comment (ptquang86) [reply src/components/core/base/GeoValidatorHOC.tsx]: updated.

File: src/execution/fixtures/ExecuteAfm.fixtures.ts (hunk 1 above)
Comment (ptquang86) [reply src/execution/fixtures/ExecuteAfm.fixtures.ts]: updated. thanks :)

File: src/execution/fixtures/ExecuteAfm.fixtures.ts (hunk 2 above)
Comment (ptquang86) [reply src/execution/fixtures/ExecuteAfm.fixtures.ts]: updated.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6413/ : SUCCESS in 9m 47s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2194/ : SUCCESS in 6s

**Comment by naibinh** [comment]
LGTM

**Comment by ptquang86** [comment]
rebase + squash only

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ptquang86** [comment]
update commit message only

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3327/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6420/ : SUCCESS in 18m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2204/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2205/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2737/ : SUCCESS in 18m 02s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3335/ : SUCCESS in 23m 41s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6430/ : SUCCESS in 20m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1543/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2211/ : SUCCESS in 34m 58s

**Pull request merged**